src/databricks/sql/cloudfetch/download_manager.py

```diff
@@ -0,0 +1,139 @@
+from collections import namedtuple
+from concurrent.futures import ThreadPoolExecutor
+from databricks.sql.cloudfetch.downloader import ResultSetDownloadHandler
+
+
+class ResultFileDownloadManager:
+
+    def __init__(self, max_download_threads, lz4_compressed):
+        self.download_handlers = []
+        self.thread_pool = ThreadPoolExecutor(max_workers=max_download_threads + 1)
+        self.downloadable_result_settings = _get_downloadable_result_settings(lz4_compressed)
+        self.fetch_need_retry = False
+        self.num_consecutive_result_file_download_retries = 0
+        self.cloud_fetch_index = 0
+
+    def add_file_links(self, t_spark_arrow_result_links, next_row_index):
+        for t_spark_arrow_result_link in t_spark_arrow_result_links:
+            if t_spark_arrow_result_link.rowCount <= 0:
+                continue
+            self.download_handlers.append(ResultSetDownloadHandler(
+                self.downloadable_result_settings, t_spark_arrow_result_link))
+        self.cloud_fetch_index = next_row_index
+
+    def get_next_downloaded_file(self, next_row_index):
```

Comment:
It's probably in a separate PR, but where is this method called from?

Reply:
It's in a separate PR, will be called from the CloudFetch queue (_create_next_table)